incubating/wrappers/java/src/main/java/io/seldon/wrapper/api/ModelPredictionController.java

```diff
@@ -11,32 +19,79 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @ConditionalOnExpression("${seldon.api.model.enabled:false}")
-public class ModelRestController {
-  private static Logger logger = LoggerFactory.getLogger(ModelRestController.class.getName());
+public class ModelPredictionController {
+  private static Logger logger = LoggerFactory.getLogger(ModelPredictionController.class.getName());
 
   @Autowired SeldonPredictionService predictionService;
 
+  /**
+   * Will accept a POST or a GET request with either a query parameter or a FORM parameter.
+   *
+   * Examples:
+   * GET -> /predict?json={ ... }
+   * curl -s \
+   *  localhost:9000/predict?json={"data": {"names": ["a", "b"], "ndarray": [[1.0, 2.0]]}}' \
+   *
+   * POST FORM -> /predict
+   * curl -s -X POST \
+   *  -d 'json={"data": {"names": ["a", "b"], "ndarray": [[1.0, 2.0]]}}' \
+   *  localhost:9000/predict
+   *
+   * @param json
+   * @return
+   * @deprecated
+   */
+  @Deprecated
```

Comment:
Since we are not doing Rest, there is no need for a call `/predict?json="{...}"` and hence deserves the deprecation tag